src/Illuminate/Cache/DatabaseStore.php

```diff
@@ -149,23 +145,32 @@ public function decrement($key, $value = 1)
      * @param  string  $key
      * @param  mixed  $value
      * @param  \Closure  $callback
-     * @return void
+     * @return int|bool
      */
     protected function incrementOrDecrement($key, $value, Closure $callback)
     {
-        $prefixed = $this->prefix.$key;
+        return $this->connection->transaction(function () use ($key, $value, $callback) {
+            $prefixed = $this->prefix.$key;
 
-        $cache = $this->table()->where('key', $prefixed)->lockForUpdate()->first();
+            $cache = $this->table()->where('key', $prefixed)->lockForUpdate()->first();
+
```

Comment:
Do you mean if `increment` or `decrement` are called simultaneously, or are you thinking of another case?